nautobot/ipam/querysets.py

```diff
@@ -191,46 +192,50 @@ def string_search(self, search):
-        last_ip = self._get_last_ip(prefix)
-        return self.filter(
-            prefix_length__gte=prefix.prefixlen,
-            network__gte=prefix.network,
-            broadcast__lte=last_ip,
-        )
-
-    def net_contains(self, prefix):
-        prefix = netaddr.IPNetwork(prefix)
-        last_ip = self._get_last_ip(prefix)
-        return self.filter(
-            prefix_length__lt=prefix.prefixlen,
-            network__lte=prefix.network,
-            broadcast__gte=last_ip,
-        )
-
-    def net_contains_or_equals(self, prefix):
-        prefix = netaddr.IPNetwork(prefix)
-        last_ip = self._get_last_ip(prefix)
-        return self.filter(
-            prefix_length__lte=prefix.prefixlen,
-            network__lte=prefix.network,
-            broadcast__gte=last_ip,
-        )
+    def net_equals(self, *prefixes):
+        query = Q()
+        for prefix in prefixes:
+            with contextlib.suppress(netaddr.AddrFormatError, ValueError):
+                prefix = netaddr.IPNetwork(prefix)
+                last_ip = self._get_last_ip(prefix)
+                query |= Q(prefix_length=prefix.prefixlen, network=prefix.network, broadcast=last_ip)
+        return self.filter(query)
+
+    def net_contained(self, *prefixes):
+        query = Q()
+        for prefix in prefixes:
+            with contextlib.suppress(netaddr.AddrFormatError, ValueError):
+                prefix = netaddr.IPNetwork(prefix)
+                last_ip = self._get_last_ip(prefix)
+                query |= Q(prefix_length__gt=prefix.prefixlen, network__gte=prefix.network, broadcast__lte=last_ip)
+        return self.filter(query)
+
+    def net_contained_or_equal(self, *prefixes):
+        query = Q()
+        for prefix in prefixes:
+            with contextlib.suppress(netaddr.AddrFormatError, ValueError):
+                prefix = netaddr.IPNetwork(prefix)
+                last_ip = self._get_last_ip(prefix)
+                query |= Q(prefix_length__gte=prefix.prefixlen, network__gte=prefix.network, broadcast__lte=last_ip)
+        return self.filter(query)
+
+    def net_contains(self, *prefixes):
+        query = Q()
+        for prefix in prefixes:
+            with contextlib.suppress(netaddr.AddrFormatError, ValueError):
+                prefix = netaddr.IPNetwork(prefix)
+                last_ip = self._get_last_ip(prefix)
+                query |= Q(prefix_length__lt=prefix.prefixlen, network__lte=prefix.network, broadcast__gte=last_ip)
+        return self.filter(query)
+
+    def net_contains_or_equals(self, *prefixes):
+        query = Q()
+        for prefix in prefixes:
+            with contextlib.suppress(netaddr.AddrFormatError, ValueError):
+                prefix = netaddr.IPNetwork(prefix)
+                last_ip = self._get_last_ip(prefix)
+                query |= Q(prefix_length__lte=prefix.prefixlen, network__lte=prefix.network, broadcast__gte=last_ip)
+        return self.filter(query)
```

Comment:
I think if we're going to update these queryset methods to accept multiple values they need to behave the same way as django. This method is always doing an OR on multiple arguments which is not how django filtering works. Instead of accepting multiple args we probably need to accept a Q object and use the deconstruct() method to pick apart the Q object and recreate it with the updated parameters.

Reply:
May need to discuss this with @glennmatthews

Reply:
We don't need to make these changes here